qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,209 @@
+        netperf_source=os.path.join(data_dir.get_deps_dir("netperf"),
+                                    params.get("netperf_server_link")),
+        password=params.get("hostpassword"),
+        compile_option=params.get("compile_option", ""))
+    wait_time = params.get_numeric("wait_bg_time")
+    ping_lost_ratio = params.get_numeric("background_ping_package_lost_ratio", 5)
+    ping_time = params.get_numeric("background_ping_time")
+    guest_ip = vm.get_address()
+    required_reboot = True
+    bg_test = True
+    try:
+        while bg_test:
+            session_serial = vm.wait_for_serial_login(timeout=login_timeout)
+            n_client.session = session_serial
+            error_context.context("Enable multi queues in guest.", logging.info)
+            for nic in vm.virtnet:
+                ifname = utils_net.get_linux_ifname(session_serial, nic.mac)
+                queues = int(nic.queues)
+                change_queues_number(ifname, queues)
+            error_context.context("Run test %s background" % netperf_stress,
+                                  logging.info)
+            stress_thread = utils_misc.InterruptedThread(netperf_test)
+            stress_thread.start()
+            utils_misc.wait_for(lambda: wait_time, 0, 5,
+                                "Wait %s start background" % netperf_stress)
+
+            # ping test
+            error_context.context("Ping guest from host", logging.info)
+            args = (guest_ip, ping_time, ping_lost_ratio)
+            bg_ping = utils_misc.InterruptedThread(ping_test, args)
+            bg_ping.start()
+
+            error_context.context("Change queues number repeatedly",
+                                  logging.info)
+            repeat_counts = params.get_numeric("repeat_counts")
+            for nic in vm.virtnet:
+                queues = int(nic.queues)
+                if queues == 1:
+                    logging.info("Nic with single queue, skip and continue")
+                    continue
+                ifname = utils_net.get_linux_ifname(session_serial, nic.mac)
+                change_list = params.get("change_list").split(",")
+                for repeat_num in range(repeat_counts):
+                    error_context.context("Change queues number -- %sth"
+                                          % repeat_num, logging.info)
+                    queues_status = get_queues_status(ifname)
+                    for q_number in change_list:
+                        queues_status = change_queues_number(ifname,
+                                                             int(q_number),
+                                                             queues_status)
+
+            logging.info("wait for background test finish")
+            try:
+                stress_thread.join()
+            except Exception as err:
+                err_msg = "Run %s test background error!\n "
+                err_msg += "Error Info: '%s'"
```

Comment:
Any particular reason to split the error message here?

Reply:
I just want the error message to displayed more clearly.